crates/control-interface/src/lib.rs

```diff
@@ -674,6 +709,51 @@ pub async fn collect_sub_timeout<T: DeserializeOwned>(
     items
 }
 
+enum IdentifierKind {
+    HostId,
+    ActorId,
+    ActorRef,
+    ProviderId,
+    ProviderRef,
+    ContractId,
+    LinkName,
+}
+
+//NOTE(ahmedtadde): For an initial implementation, we just want to make sure that the identifier is, at very least, not an empty string.
+//This parser should be refined over time as needed.
+fn parse_identifier<T: AsRef<str>>(kind: &IdentifierKind, value: T) -> Result<String> {
+    let value = value.as_ref();
+    let assert_non_empty_string = |input: &str, message: Option<String>| -> Result<String> {
```

Comment:
Nit: Can we just break this out into an actual function rather than an enclosure? Also, since this is a private function and we are always passing in a message context, let's just remove the `Option<String>` and just make it a `&str` (which also avoids the allocation)